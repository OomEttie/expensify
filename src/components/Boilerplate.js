import React from 'react';
import Header from './Header';

export default class BoilerplateApp extends React.Component {
  state = {
  };

  //FUNCTIONS
  render() {
    return (
      <div>
        <Header />
      </div>
    );
  }
}

// 
// REDUX CONNECTED
//
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export class LoginPage extends React.Component {
  render() {
    return (
      <div>
        <p>This is the Login Page</p>
        <Link to="/dashboard">go to dashboard!</Link>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {};
};

export default connect(mapStateToProps)(LoginPage);
