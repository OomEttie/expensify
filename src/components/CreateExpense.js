import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';
// onSubmit = expense => {
//   console.log('onSubmit-', expense);
// };

export class CreateExpense extends React.Component {
  onSubmit = expense => {
    // props.dispatch(addExpense(expense));
    this.props.createExpense(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <h3>Add Expense</h3>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    createExpense: expense => {
      dispatch(addExpense(expense));
    }
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(CreateExpense);
