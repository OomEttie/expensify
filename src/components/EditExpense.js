import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpense extends React.Component {
  onSubmit = expense => {
    // props.dispatch(addExpense(expense));
    this.props.startEditExpense(this.props.expense.id, { ...expense });
    this.props.history.push('/');
  };

  onClick = () => {
    // console.log(dispatch);
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className="button__warning" onClick={this.onClick}>REMOVE</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  // console.log('mapStateToProps-props->', props)
  return {
    expense: state.expenses.find(expense => {
      return expense.id == props.match.params.id;
    })
  };
};

const mapDispatchToProps = dispatch => {
  return {
    startEditExpense: (id, updates) => {
      dispatch(startEditExpense(id, updates));
    },
    startRemoveExpense: id => {
      dispatch(startRemoveExpense(id));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpense);
