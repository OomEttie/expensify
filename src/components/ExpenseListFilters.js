import React from 'react';
import { connect } from 'react-redux';
import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate
} from '../actions/filters';

import { DateRangePicker } from 'react-dates';
import 'react-dates/initialize';
// import 'react-dates/lib/css/_datepicker.css';

class ExpenseListFilters extends React.Component {
  state = {
    calenderFocus: null
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChange = calenderFocus => {
    this.setState(() => ({ calenderFocus: calenderFocus }));
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={e => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={e => {
            if (e.target.value == 'date') {
              this.props.dispatch(sortByDate());
            } else {
              this.props.dispatch(sortByAmount());
            }
          }}
        >
          <option value="date">Date Desc</option>
          <option value="amount">Amount Desc</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
          endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocus}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
}

const MapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(MapStateToProps)(ExpenseListFilters);
