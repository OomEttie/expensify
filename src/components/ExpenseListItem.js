import React from 'react';
import { NavLink } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({
  dispatch,
  id,
  description,
  amount,
  createdAt
}) => (
  <div>
    <NavLink exact to={`/edit/${id}`}>
      <h2>{description}</h2>
    </NavLink>
    <p>Amount: {numeral(amount / 100).format('$0,0.00')}</p>
    <p>Created At: {moment(createdAt).format('MMM Do, YYYY')}</p>
  </div>
);

// moment.unix(createdAt).toString("MMM DD, YYYY")
// export const ExpenseListItem = connect()(ExpenseListItemConnected);

// export default ExpenseListItem;
// export default connect()(ExpenseListItem);
