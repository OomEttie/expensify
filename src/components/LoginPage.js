import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginPage extends React.Component {
  onClick = () => {
    this.props.startLogin();
  };

  render() {
    return (
      <div>
        <p>This is the Login Page</p>
        <button onClick={this.onClick}>Login</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {};
};

const mapDispatchToProps = dispatch => {
  return {
    startLogin: () => {
      dispatch(startLogin());
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginPage);
