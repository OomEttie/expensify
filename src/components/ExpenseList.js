import React from 'react';
import { connect } from 'react-redux';
import { ExpenseListItem } from './ExpenseListItem';
import { getVisibleExpenses } from '../selectors/expenses';

// <h1>Expense List</h1>
export const ExpenseList = props => (
  <div>
    {props.expenses.length == 0 ? (
      <p>No Expenses</p>
    ) : (
      props.expenses.map(expense => {
        // console.log(expense);
        // const Temp = () => (
        //   <div>
        //     <p>Amount: </p>
        //     <p>Created At: </p>
        //   </div>
        // )
        // return <Temp />;
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList);

// export default ExpenseList;
export default ConnectedExpenseList;
