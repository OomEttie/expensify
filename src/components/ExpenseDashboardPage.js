import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

export const ExpenseDashboardPage = () => (
  <div>
    <h3>This is my expense Dashboard component!</h3>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
