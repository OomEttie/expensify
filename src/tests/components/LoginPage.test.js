import React from 'react';
import { shallow } from 'enzyme';

import { LoginPage } from '../../components/LoginPage';

let startLogin;

beforeAll(() => {
  startLogin = jest.fn();
});

test('render LoginPage component', () => {
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  expect(wrapper).toMatchSnapshot();
});


test('should call startLogin on button click', () => {
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').prop('onClick')();

  expect(startLogin).toHaveBeenCalled();
});