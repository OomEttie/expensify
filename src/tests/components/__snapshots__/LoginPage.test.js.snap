// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render LoginPage component 1`] = `
<div>
  <p>
    This is the Login Page
  </p>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
